### PR TITLE
feat: Added support for single-core applications and optional elf file

### LINKED ISSUE #11: Make elf path optional when listing devices
Currently the user must supply an elf path when listing devices. This argument is actually not used, the tool will exit after listing devices and can actually be a path _not_ pointing to a file.

See https://github.com/therealfrauholle/tricore-probe/commit/f95ac870bedf0c93e43d6e6790283cf8c9cb8c38 for a reference, e.g. a hack that could solve this. This could already solve the problem, but we want to use clap to its full potential to nicely reflect the usage of elf path in the CLI.

### LINKED ISSUE #36: Make configurable the exit condition of tricore-probe.
We are using tricore-probe to run bare-metal code that use only one core.
tricore-probe exit immediately because it check when any core is halted and therefore we cannot see any log message.
I suggest to make the exit condition configurable.
I propose to add a command line option like that:
_--exit-condition_

Values:
* all_halted
* any_halted (default)
* core0_halted
* ....<all other core>

### PR DESCRIPTION
**Pull request by tommasoprandin**
This PR includes support for not specifying the .elf file on operations that don't require it (eg. `--list-devices`) and for limiting the number of cores checked for session termination.
In this way it is possible to see `defmt` messages on single (or dual) core applications without the session terminating abruptly.

The flag `--cores` has been added to the CLI in order to specify the number of active cores.

NOTE: It was necessary to revert to v1.0.10 of `AURIXFlasher` and to version `8.0.5`, otherwise the board was not being recognised (Shieldbuddy TC375)

resolves #36, resolves #11

### CODE REVIEW

File: src/defmt.rs [hunk 1]
```diff
@@ -244,6 +246,15 @@ pub fn decode_rtt<W: Write>(
...
         for (secondary_index, core) in secondary_cores.iter_mut().enumerate() {
+            log::debug!("Secondary core index: {}, active cores: {}", secondary_index, active_cores);
+            // eww
+            let active_secondary_cores = (active_cores - 1).into();
+            let secondary_core_id = secondary_index + 1;
+            if secondary_core_id > active_secondary_cores {
+                // Core is not active so ignore it and all the next ones
+                log::debug!("Breaking away");
+                break;
+            }
```
Comment (arctic-alpaca) [inline src/defmt.rs]: Please replace this with [`Iterator::take()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take).

File: src/defmt.rs [hunk 2]
```diff
@@ -244,6 +246,15 @@ pub fn decode_rtt<W: Write>(
         }
 
         for (secondary_index, core) in secondary_cores.iter_mut().enumerate() {
+            log::debug!("Secondary core index: {}, active cores: {}", secondary_index, active_cores);
```
Comment (arctic-alpaca) [inline src/defmt.rs]:
```suggestion
            log::debug!("Secondary core index: {}, active cores: {}", secondary_index + 1, active_cores);
```

This should log the same index as the log at https://github.com/veecle/tricore-probe/pull/37/files#diff-076d741b15a7f8c1976902c1b98e7ab83d29519ef4385c3ac6c758bbdf4b0a4eR262-R266

File: README.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ docker build . --tag veecle/flash-tricore --build-arg=AGREE_INFINEON_TERMS=1 -f
 
 Install `tricore-probe`:
 ```shell
-cargo install tricore-probe --git https://github.com/veecle/tricore-probe --version 0.2.0
+cargo install --path .
```
Comment (arctic-alpaca) [inline README.md]: Please revert this change, this is intended for people installing the latest released version.

**Review by arctic-alpaca — COMMENTED** [review]
Thank you!

Could you make the default behavior for the `cores` CLI parameter to use all cores (same as current `main` behavior)?

For boards with three cores, if I were to use core 0 and core 2, the current approach would run into limitations (assuming the secondary cores are ordered). Could you document this limitation?

**Comment by arctic-alpaca** [comment]
> NOTE: It was necessary to revert to v1.0.10 of AURIXFlasher and to version 8.0.5, otherwise the board was not being recognised (Shieldbuddy TC375)

We don't have the hardware to reproduce the issue. If this is also the case when using the AURIX Flasher tool without tricore-probe, it might be worth it to report this to Infineon.

File: src/defmt.rs (hunk 2 above)
Comment (arctic-alpaca) [reply src/defmt.rs]: On second thought, please remove that log line. Logging this on every iteration makes debug-logs very hard to read.

File: src/defmt.rs (hunk 1 above)
Comment (tommasoprandin) [reply src/defmt.rs]: I have implemented the change. I still feel that, considering the code examples available, it is more desirable to have only one core by default, but I guess for backward compatibility it is better to leave things like that.

File: README.md (hunk 3 above)
Comment (tommasoprandin) [reply README.md]: I have fixed this, I made a version change to 0.2.1, I can change that if needed

**Review by arctic-alpaca — APPROVED** [review]
Thanks!
